src/util/child_common.c

```diff
@@ -488,6 +490,7 @@ static struct tevent_req *_read_pipe_send(TALLOC_CTX *mem_ctx,
     state->buf = NULL;
     state->len = 0;
     state->safe = safe;
+    state->non_blocking = non_blocking;
```

Comment:
If I understand correctly, combination `((safe == true) && (non_blocking == true))` doesn't make sense.
Would you mind to add a sanity check to make sure potential misuse is caught?